ThirdPartyModules/Aheadworks/StoreCredit.php

```diff
@@ -98,4 +107,53 @@ public function collectDiscounts(
             return [$discounts, $totalAmount, $diff];
         }
     }
+    
+    /**
+     * Return code if the quote has Aheadworks store credits.
+     * 
+     * @param $result
+     * @param $couponCode
+     * @param $quote
+     * 
+     * @return array
+     */
+    public function filterVerifyAppliedStoreCredit (
+        $result,
+        $couponCode,
+        $quote
+    )
+    {
+        if ($couponCode == self::AHEADWORKS_STORE_CREDIT && $quote->getAwUseStoreCredit()) {
+            $result[] = $couponCode;
+        }
+        
+        return $result;
+    }
+    
+    /**
+     * Remove Aheadworks store credits from the quote.
+     *
+     * @param $amastyApplyStoreCreditToQuote
+     * @param $couponCode
+     * @param $quote
+     * @param $websiteId
+     * @param $storeId
+     * 
+     */
+    public function removeAppliedStoreCredit (
+        $couponCode,
+        $quote,
+        $websiteId,
+        $storeId
+    )
+    {
+        try {
+            if ($couponCode == self::AHEADWORKS_STORE_CREDIT && $quote->getAwUseStoreCredit()) {
+                $quote->setAwUseStoreCredit(false);
+                $this->quoteRepository->save($quote->collectTotals());
+            }
+        } catch (\Exception $e) {
+            throw $e;
```

Comment:
@dumega Can you log the exception to Bugsnag?

Reply:
@johnmccombs1 it would sent to bugsnag via https://github.com/BoltApp/bolt-magento2/blob/master/Model/Api/UpdateDiscountTrait.php#L474